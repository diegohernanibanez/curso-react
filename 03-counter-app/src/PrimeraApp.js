// import React, { Fragment } from 'react';

// const PrimeraApp = () => {
//     return (
//         <Fragment>
//             <h1>Hola Mundo</h1>
//             <p>Mi primera app de React</p>
//         </Fragment>
//     );
// }

import React from 'react';
import PropTypes from 'prop-types';

const PrimeraApp = ({saludo}) => {
    
    // const numeros = 123;
    // const decimales = 123.123;
    // const boolean = true;
    // const arreglo = [1, 2, 3, 4, 5, 6];
    // const objeto = {
    //     nombre: 'Diego',
    //     edad: 26,
    // }
    //Se puede retornar con <div></div> o se puede retornar con <></>
    return (
        <>
            <h1>{saludo}</h1>
            <p>Mi primera app de React</p>
            {/* <h1>{numeros}</h1>
            <h1>{decimales}</h1>
            <h1>{boolean}</h1>
            <h1>{arreglo}</h1>
            <pre>{JSON.stringify(objeto, null, 3)}</pre> */}
        </>
    );
}

PrimeraApp.propTypes = {
    saludo: PropTypes.string.isRequired,
}

export default PrimeraApp;