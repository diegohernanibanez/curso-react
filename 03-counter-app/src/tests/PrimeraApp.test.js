import React from 'react';
// import {render} from "@testing-library/react";
import PrimeraApp from "../PrimeraApp";
import {shallow} from 'enzyme';

describe('Pruena en PrimeraApp', () => {
    // test('debe mostrar "Hola Mundo"', () => {
    //     const saludo = 'Hola Mundo';

    //     const {getByText} = render(<PrimeraApp saludo = {saludo} />);
        
    //     expect(getByText(saludo)).toBeInTheDocument();
    // })
    
    test('debe mostrar PrimeraApp correctamente', () => {
        const saludo = 'Hola Mundo';
        const wrapper = shallow(<PrimeraApp saludo = {saludo} />);


        expect(wrapper).toMatchSnapshot();
    })
    

})
