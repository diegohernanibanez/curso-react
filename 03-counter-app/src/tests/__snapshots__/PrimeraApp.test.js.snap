// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruena en PrimeraApp debe mostrar PrimeraApp correctamente 1`] = `
<Fragment>
  <h1>
    Hola Mundo
  </h1>
  <p>
    Mi primera app
  </p>
</Fragment>
`;
